central/convert/storagetov1/cloud_source.go

```diff
@@ -0,0 +1,36 @@
+package storagetov1
+
+import (
+	v1 "github.com/stackrox/rox/generated/api/v1"
+	"github.com/stackrox/rox/generated/storage"
+)
+
+// CloudSource converts the given storage.CloudSource to a v1.CloudSource.
+func CloudSource(cloudSource *storage.CloudSource) *v1.CloudSource {
+	v1CloudSource := &v1.CloudSource{
+		Id:                  cloudSource.GetId(),
+		Name:                cloudSource.GetName(),
+		Type:                cloudSourceToV1TypeEnum(cloudSource.GetType()),
+		SkipTestIntegration: cloudSource.GetSkipTestIntegration(),
```

Comment:
So we effectively dropped the `Credentials` message now completely and do not explicitly do a scrub call, not sure this plays well with the UI expecting this field to be set and "clearing" the scrubbed values (if you look at the logic for existing integrations, the UI expected the password fields to have "*****" and attempts to clear it).

On a more general concept: I'm not sure I like the indirection on dropping the secret value within the conversion layer. In case of a new field being added which is a secret, this would have to be explicitly maintained within the conversion layer instead of simply calling the scrub functions here.

So, let's not introduce the indirection and have the conversion layer also have a call to the scrub layer, this way we can be sure that every field that's supposed to be scrubbed is actually scrubbed based on the tag / proto definition, and there's less room for potentially leaking secrets in the future when new fields are added.